src/main/java/com/freshman/freshmanbackend/domain/member/controller/MemberController.java

```diff
@@ -0,0 +1,48 @@
+package com.freshman.freshmanbackend.domain.member.controller;
+
+import com.freshman.freshmanbackend.domain.member.controller.validator.MemberValidator;
+import com.freshman.freshmanbackend.domain.member.request.MemberInfoUpdateRequest;
+import com.freshman.freshmanbackend.domain.member.response.MemberInfoResponse;
+import com.freshman.freshmanbackend.domain.member.service.MemberService;
+import com.freshman.freshmanbackend.global.common.response.DataResponse;
+import com.freshman.freshmanbackend.global.common.response.SuccessResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+
+/**
+ * 멤버 컨트롤러
+ */
+@RestController
+@RequiredArgsConstructor
+public class MemberController {
+    private final MemberService memberService;
+
+    @PostMapping("/user")
+    public ResponseEntity<?> saveMemberInfo(@RequestBody @Validated MemberInfoUpdateRequest memberInfoUpdateRequest, Principal principal){
+        System.out.println("saving!");
+        memberService.saveMemberInfo(memberInfoUpdateRequest, principal.getName());
+        return ResponseEntity.ok(new SuccessResponse());
+    }
+
+    @GetMapping("/user")
+    public ResponseEntity<?> getMemberInfo(Principal principal){
+        MemberInfoResponse memberInfo = memberService.getMemberInfo(principal.getName());
+        return ResponseEntity.ok(new DataResponse(memberInfo));
+    }
```

Comment:
```suggestion
@RestController
@RequiredArgsConstructor
@RequestMapping("/products")
public class MemberController {
    private final MemberService memberService;

    @PostMapping
    public ResponseEntity<?> saveMemberInfo(@RequestBody @Validated MemberInfoUpdateRequest memberInfoUpdateRequest, Principal principal){

        memberService.saveMemberInfo(memberInfoUpdateRequest, principal.getName());
        return ResponseEntity.ok(new SuccessResponse());
    }

    @GetMapping
    public ResponseEntity<?> getMemberInfo(Principal principal){
        MemberInfoResponse memberInfo = memberService.getMemberInfo(principal.getName());
        return ResponseEntity.ok(new DataResponse(memberInfo));
    }
```

* RequestMapping 어노테이션을 통해 공통 경로를 간결하게 설정해 줄 수 있습니다!
* 테이블 명과 엔티티 클래스명도 Member인데 경로가 user로 설정하신 이유가 있으실까요?? 그리고 엔티티명을 경로에 사용할 때는 복수형 단어로 사용하는게 rest api 규칙에 더 맞는 것 같습니다.

Reply:
- 앗 ERD설계하기 전에 api 엔드포인트를 만들었는데 그 이후에 반영을 안했네요.
- 헤더로 사용자 정보가 넘어가서 결국 가리키는 것은 하나이기 때문에 컬렉션 리소스가 아니라 도큐먼트 리소스라고 생각했습니다. 하지만 다시 생각해보니 컬렉션 리소스가 맞는 것 같네요!